### PR TITLE
Add support for translation for the sample Dell BIOS setting strings

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

File: po/POTFILES.in [hunk 1]
```diff
@@ -1,5 +1,6 @@
 data/remotes.d/lvfs.metainfo.xml
 data/remotes.d/lvfs-testing.metainfo.xml
+libfwupdplugin/tests/bios-attrs/dell-xps13-9310/strings.txt
```
Comment (CragW) [inline po/POTFILES.in:3]: should we check this txt file?
https://github.com/fwupd/fwupd/blob/main/contrib/ci/check-potfiles.py#L20

Comment (superm1) [reply po/POTFILES.in:3]: thx, added it.

**Review by CragW — DISMISSED** [review]

File: contrib/ci/populate-bios-attr-translations.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+import sys
+
+
+def populate_translations(path):
+    output = open(os.path.join(path, "strings.txt"), "w")
+    for root, dirs, files in os.walk(path):
```
Comment (hughsie) [inline contrib/ci/populate-bios-attr-translations.py:19]: yup, `for root, _, files in os.walk(path):` perhaps?

File: contrib/ci/populate-bios-attr-translations.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        for file in files:
+            val = ""
```
Comment (hughsie) [inline contrib/ci/populate-bios-attr-translations.py:21]: `val: str = ""` perhaps?

File: libfwupdplugin/fu-bios-attrs.c [hunk 4]
```diff
@@ -368,6 +368,10 @@ fu_bios_attrs_setup(FuBiosAttrs *self, GError **error)
 		if (driver == NULL)
 			break;
 		path = g_build_filename(sysfsfwdir, driver, "attributes", NULL);
+		if (!g_file_test(path, G_FILE_TEST_IS_DIR)) {
+			g_debug("Skipping non-directory %s", path);
```
Comment (hughsie) [inline libfwupdplugin/fu-bios-attrs.c:372]: lowercase `Skipping` pls.

File: src/fu-util-bios-attr.c [hunk 5]
```diff
@@ -36,6 +36,9 @@ fu_util_bios_attr_update_description(FwupdBiosAttr *attr)
 			/* TRANSLATORS: description of a BIOS setting */
 			new = _("BIOS updates delivered via LVFS or Windows Update");
 		}
+		/* Try to look it up from translations */
+	} else {
+		new = _(old);
```
Comment (hughsie) [inline src/fu-util-bios-attr.c:41]: I'm not sure this is 100% correct; don't you want a call to `gettext()` instead?

File: src/fu-util-bios-attr.c [hunk 6]
```diff
@@ -36,6 +36,9 @@ fu_util_bios_attr_update_description(FwupdBiosAttr *attr)
 			/* TRANSLATORS: description of a BIOS setting */
 			new = _("BIOS updates delivered via LVFS or Windows Update");
 		}
+		/* Try to look it up from translations */
```
Comment (hughsie) [inline src/fu-util-bios-attr.c:39]: lowercase `Try` and maybe move to above the _()?

File: libfwupdplugin/tests/bios-attrs/dell-xps13-9310/strings.txt [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+Service Tag
+
+#TRANSLATORS: Description of BIOS setting
+Keyboard Illumination
+
+#TRANSLATORS: Description of BIOS setting
+Intel Speed Shift Technology
+
+#TRANSLATORS: Description of BIOS setting
+
```
Comment (hughsie) [inline libfwupdplugin/tests/bios-attrs/dell-xps13-9310/strings.txt:11]: ??

**Review by hughsie — APPROVED** [review]

**Pull request merged**